v2/internal/test/fakeworkloadapi/workload_api_windows.go

```diff
@@ -0,0 +1,39 @@
+//go:build windows
+// +build windows
+
+package fakeworkloadapi
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/spiffe/go-spiffe/v2/proto/spiffe/workload"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+func NewWithNamedPipeListener(tb testing.TB) *WorkloadAPI {
+	w := &WorkloadAPI{
+		x509Chans:       make(map[chan *workload.X509SVIDResponse]struct{}),
+		jwtBundlesChans: make(map[chan *workload.JWTBundlesResponse]struct{}),
+	}
+
+	listener, err := winio.ListenPipe(fmt.Sprintf(`\\.\pipe\go-spiffe-test-pipe-%x`, rand.Uint64()), nil)
```

Comment:
nitpick: If we don't seed rand, then two test processes running concurrently could collide on the names :) We're more than likely ok, but just thought I'd point it out just in case.